shell/components/CommunityLinks.vue

```diff
@@ -63,12 +63,20 @@ export default {
       }
 
       // Custom links set from settings
-      if (this.uiCustomLinks) {
-        return JSON.parse(this.uiCustomLinks.value).reduce((prev, curr) => {
-          prev[curr.key] = curr.value;
-
-          return prev;
-        }, {});
+      if (this.uiCustomLinks?.value) {
+        try {
+          const customLinks = JSON.parse(this.uiCustomLinks.value);
+
+          if (Array.isArray(customLinks)) {
+            return JSON.parse(this.uiCustomLinks.value).reduce((prev, curr) => {
+              prev[curr.key] = curr.value;
+
+              return prev;
+            }, {});
+          }
+        } catch (e) {
+          console.error('Could not parse custom links setting', e); // eslint-disable-line no-console
+        }
```

Comment:
```suggestion
        if (!!this.uiCustomLinks?.value) {
        const val = JSON.parse(this.uiCustomLinks.value);

       // User could have no links.
        const customLinks = Array.isArray(val) ? val : Object.keys({});

        return customLinks.reduce((prev, curr) => {
           prev[curr.key] = curr.value;

          return prev;
        }, {});
      } 
```

Reply:
Unless I'm missing something here I think `.reduce` is fine on an empty array if there's an initial value set? Other than my uncertainty around this the PR looks good.